eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -95,11 +95,41 @@ class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient) extends OnChainWall
       index = txs.indexOf(JString(txid.toHex))
     } yield (BlockHeight(height.toInt), index)
 
+  /**
+   * Return true if this output can potentially be spent.
+   *
+   * Note that if this function returns false, that doesn't mean the output cannot be spent. The output could be unknown
+   * (not in the blockchain nor in the mempool) but could reappear later and be spendable at that point. If you want to
+   * ensure that an output is not spendable anymore, you should use [[isTransactionOutputSpent]].
+   */
   def isTransactionOutputSpendable(txid: ByteVector32, outputIndex: Int, includeMempool: Boolean)(implicit ec: ExecutionContext): Future[Boolean] =
     for {
       json <- rpcClient.invoke("gettxout", txid, outputIndex, includeMempool)
     } yield json != JNull
 
+  /**
+   * Return true if this output has already been spent by a confirmed transaction.
+   * Note that a reorg may invalidate the result of this function and make a spent output spendable again.
+   */
+  def isTransactionOutputSpent(txid: ByteVector32, outputIndex: Int)(implicit ec: ExecutionContext): Future[Boolean] = {
+    getTxConfirmations(txid).flatMap {
+      case Some(confirmations) if confirmations > 0 =>
```

Comment:
I'm not sure this adds anything since later we'll check if `txid:outputIndex` is a confirmed UTXO ?
And the test passes even with the old version of `doubleSpent()`

Reply:
> And the test passes even with the old version of doubleSpent()

`detect if tx has been double-spent` passes with the old version of `doubleSpent()` (because it has been refactored but not changed to highlight the bug) but `detect if tx has been double-spent (with unconfirmed inputs)` does *not* pass with the old version of `doubleSpent()`.

> I'm not sure this adds anything since later we'll check if txid:outputIndex is a confirmed UTXO ?

What do you mean? Where do we check that "txid:outputIndex is a confirmed UTXO"? It is absolutely necessary, this function by definition checks "if this output has already been spent by a confirmed transaction."

Without this check this function would return `true` for unconfirmed inputs, which is incorrect.

Reply:
Ok so it seems that I was completely wrong, misunderstood the PR was and somehow did not run the tests correctly :), nice fix!